trade-finance-manager-api/src/v1/controllers/deal.submit.controller.js

```diff
@@ -144,14 +149,33 @@ const submitDealAfterUkefIds = async (dealId, dealType, checker) => {
 
 exports.submitDealAfterUkefIds = submitDealAfterUkefIds;
 
+const dealHasAllUkefId = async (dealId) => {
+  const deal = await findOneTfmDeal(dealId);
+
+  if (deal && deal.dealSnapshot && deal.dealSnapshot.facilities) {
+    const dealHasId = !!deal.dealSnapshot.ukefDealId;
+    const facilitiesHaveIds = deal.dealSnapshot.facilities.filter((f) => !f.facilitySnapshot.ukefFacilityId).length === 0;
+    return dealHasId && facilitiesHaveIds;
+  }
+
+  return false;
+};
+
+exports.dealHasAllUkefId = dealHasAllUkefId;
+
 const submitDealPUT = async (req, res) => {
   const {
     dealId,
     dealType,
     checker,
   } = req.body;
+  let deal;
 
-  const deal = await submitDealBeforeUkefIds(dealId, dealType, checker);
+  if (await dealHasAllUkefId(dealId)) {
+    deal = await submitDealAfterUkefIds(dealId, dealType, checker);
```

Comment:
Do you think it'd be easier on the eye like this?

```js
const canSubmitDealAfterUkefIds = await dealHasAllUkefId(dealId);

if (canSubmitDealAfterUkefIds) {
 // ...
} else {
 // ...
}
```

Just my personal preference though 🤷

Reply:
Good shout, have updated.